skops/hub_utils/_hf_hub.py

```diff
@@ -84,12 +176,29 @@ def recursively_default_dict():
     config = recursively_default_dict()
     config["sklearn"]["model"]["file"] = model_path
     config["sklearn"]["environment"] = requirements
+    config["sklearn"]["task"] = task
+
+    if "tabular" in task:
+        config["sklearn"]["example_input"] = _get_example_input(data)
+        config["sklearn"]["columns"] = _get_column_dtypes(data)
+    elif "text" in task:
+        if isinstance(data, list):
+            config["sklearn"]["example_input"] = {"data": data[:3]}
+        else:
+            raise ValueError("The data needs to be a list of strings.")
 
     with open(Path(dst) / "config.json", mode="w") as f:
         json.dump(config, f, sort_keys=True, indent=4)
 
 
-def init(*, model: Union[str, Path], requirements: List[str], dst: Union[str, Path]):
+def init(
+    *,
+    model: Union[str, Path],
+    requirements: List[str],
+    dst: Union[str, Path],
+    task: str,
+    data,
```

Comment:
I'm not quite certain. Is `data` allowed to be None or not? From the docstring, it seems to be not, but in the tests you pass `data=None`.

Reply:
Yeah the tests aren't supposed to be passing, I haven't done the tests yet. Was just wondering if you think the API is okay.